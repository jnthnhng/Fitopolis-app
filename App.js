import { StatusBar } from "expo-status-bar";
import React, { useState } from "react";
import { StyleSheet, Text, View, Button, TouchableOpacity } from "react-native";
import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import LoginScreen from "./screens/LoginScreen";
import HomeScreen from "./screens/HomeScreen";

// Source: https://everyday.codes/react-native/iterate-faster-with-github-actions-for-react-native/
// Used to help set up app with jest for CI

const Stack = createNativeStackNavigator();

export default function App() {
  const [counter, setCounter] = useState(0);
  return (
    <>
      <StatusBar style="light" />
      <NavigationContainer>
        <View style={styles.container}>
          <Text style={styles.logo}>Fitopolis</Text>
          <TouchableOpacity style={styles.button}>
            <Text style={styles.buttonText}>Login</Text>
          </TouchableOpacity>
          <TouchableOpacity style={styles.button}>
            <Text style={styles.buttonText}>Register</Text>
          </TouchableOpacity>
        </View>
      </NavigationContainer>
    </>
  );
}

const styles = StyleSheet.create({
  label: {
    marginBottom: 16,
    marginTop: 16,
  },
  container: {
    flex: 1,
    backgroundColor: "#e6e4df",
    alignItems: "center",
    justifyContent: "center",
  },
  logo: {
    color: "black",
    fontWeight: "bold",
    fontSize: 50,
    marginBottom: 30,
  },
  button: {
    paddingTop: 10,
    paddingBottom: 10,
    marginTop: 10,
    backgroundColor: "#3b3a39",
    borderRadius: 10,
    width: "60%",
    justifyContent: "center",
    alignItems: "center",
  },
  buttonText: {
    color: "white",
    fontSize: 20,
  },
});
