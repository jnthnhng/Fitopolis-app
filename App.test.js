import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react-native'
import App from './App';

// Source: https://everyday.codes/react-native/iterate-faster-with-github-actions-for-react-native/



afterEach(cleanup);

describe('<App />', () => {
  it('should match snapshot', () => {
    const rendered = render(<App />).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it('should set counter to 0', () => {
    const rendered = render(<App />);
    const counterComponent = rendered.getByTestId('counter');

    expect(counterComponent.props.children).toContainEqual(0);
  });

  it('should increase counter by 1', () => {
    const rendered = render(<App />);
    const counterComponent = rendered.getByTestId('counter');
    const buttonComponent = rendered. getByTestId('button');

    fireEvent(buttonComponent, 'press');

    expect(counterComponent.props.children).toContainEqual(1);
  });
});