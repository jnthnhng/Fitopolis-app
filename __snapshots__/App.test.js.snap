// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "marginBottom": 16,
      }
    }
    testID="counter"
  >
    You clicked the button 
    0
     times.
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="button"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press me
      </Text>
    </View>
  </View>
</View>
`;
